packages/fuels/src/cli/commands/dev/index.ts

```diff
@@ -0,0 +1,64 @@
+import * as chokidar from 'chokidar';
+import { globSync } from 'glob';
+
+import type { FuelsConfig } from '../../types';
+import { error, log } from '../../utils/logger';
+import { build } from '../build';
+import { deploy } from '../deploy';
+
+import { startFuelCore } from './startFuelCore';
+
+export const fileHandlers: {
+  watcher?: chokidar.FSWatcher;
+} = {};
+
+export const buildAndDeploy = async (config: FuelsConfig) => {
+  await build(config);
+  return deploy(config);
+};
+
+export const changeListener = (config: FuelsConfig) => async (path: string) => {
+  log(`\nFile changed: ${path}`);
+  await buildAndDeploy(config);
+};
+
+export async function dev(config: FuelsConfig) {
+  // here we inject a `providerUrl` into the config if necessary
+  if (config.autoStartFuelCore) {
+    const client = await startFuelCore(config);
+    // eslint-disable-next-line no-param-reassign
+    config.providerUrl = client.providerUrl;
+  }
+
+  const { contracts, scripts, predicates, chainConfig, workspace } = config;
+
+  const projectDirs = [contracts, scripts, predicates, chainConfig].flat();
```

Comment:
`chainConfig` is added here but from what I understand it's not watched in `pathsToWatch` because of the `globSync`s. Am I missing something?

Reply:
Good catch! 🕵️‍♂️ 

I will fix it, and unlike other files, this will require restarting the `fuel-core` node.

Perhaps a similar measure could be done for `fuels.config.ts`.